### ISSUE TITLE
Reading knownhosts file bug

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5616]
|<img alt="termim's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| termim reported|
|-|-|
|Trac ID|trac#5616|
|Type|defect|
|Created|2012-04-06 04:29:43Z|

Reading knownhosts file should not fail if there is an unknown or broken hashed entry present in the file. The behavior should be the same as in case of plain entry. The patch attached fixes this bug.

Attachments:

* [knownhosts.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ec2/ec2596ee398842aa29545dae966a00dccc087660/36ae1a1a681d53c312e590d807013a9439c7b37e.patch) (1916 bytes) - added by termim on 2012-04-06 04:31:15Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5616 5616
type__defect defect
reporter__termim termim
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1333686583000000 1333686583000000
changetime__1333732878000000 1333732878000000
version__None None
owner__exarkun exarkun
cc__termim cc__z3p
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

Thanks!  I think the unit tests for this code could be better factored, but it was broken when you got here, so I'm not going to hold this ticket up on improving that situation. :)  I did file [#5617](https://github.com/twisted/twisted/issues/5617) though, so hopefully someone will look at that someday.

The change itself looks fine, as does the test (apart from my comment above).  One thing that's missing is a [ReviewProcess#newsfiles news fragment].  I'll add one for this fix, though, and then apply the patch.  Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [34101]) Add error handling for parsing of hashed entrieds in a known_hosts file.

Author: termim
Reviewer: exarkun
Fixes: [#5616](https://github.com/twisted/twisted/issues/5616)

If an unparseable line with a hashed entry prefix is found in the known hosts file,
create an UnparsedEntry for it, just as is done for non-hashed entries.
